### PR TITLE
Restrict allowed schemas for "Open Workspace"

### LINKED ISSUE #127867: File\Open Workspace... uses remote file open dialog when currently open workspace has root(s) using a FileSystemProvider scheme
... it's still the case that when I use `File\Open Workspace...` when I already have a workspace open that uses my FSP's scheme, the remote file open dialog opens. And if my remote is now unavailable this dialog hangs for a long time before allowing me to "Show Local".

_Originally posted by @gjsjohnmurray in https://github.com/microsoft/vscode/issues/127647#issuecomment-871490234_

### PR DESCRIPTION
**Pull request by alexr00**
Fixes #127867

### CODE REVIEW

**Review by bpasero — COMMENTED** [review]
Would it make sense to do this from within the `pickWorkspaceAndOpen` method? I.e. if no `availableFileSystems` is provided from the outside, create an array with `file` and `vscodeRemote` (if remote authority is set)?

**Comment by alexr00** [comment]
Yes, it could be moved there. I just liked the symmetry since for `pickNewWorkspacePath` it is done outside of the file dialog service.

File: src/vs/workbench/browser/actions/workspaceCommands.ts [hunk 1]
```diff
@@ -54,7 +54,9 @@ CommandsRegistry.registerCommand({
 
 CommandsRegistry.registerCommand({
 	id: 'workbench.action.openWorkspaceInNewWindow',
-	handler: (accessor: ServicesAccessor) => accessor.get(IFileDialogService).pickWorkspaceAndOpen({ forceNewWindow: true })
+	handler: (accessor: ServicesAccessor) => {
```
Comment (bpasero) [inline src/vs/workbench/browser/actions/workspaceCommands.ts:57]: This change is not needed I think.

**Review by bpasero — CHANGES_REQUESTED** [review]
I am not entirely sure why the new `this.getWorkspaceAvailableFileSystems` method is limited to `pickWorkspaceAndOpen` only, is it not equally warranted for `pickFolderAndOpen` and/or `pickFileAndOpen`?

I think I am missing the background of the issue here why the fix is only needed in that one method, maybe that could be explained further.

**Comment by alexr00** [comment]
@bpasero it is only needed for workspace files because workspace files cannot be read from file system provider provided file systems (https://github.com/microsoft/vscode/issues/126476#issuecomment-863170411). It is still perfectly valid to open a folder or file from a virtual file system.

**Review by bpasero — APPROVED** [review]

**Pull request merged**